### PR TITLE
Public provider curation api docs

### PR DESCRIPTION
**Pull request by bethanyr**
## Labels

- [ ] inaccuracy
- [ ] clarification
- [x] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout
- [ ] api - requires an update to the [changelog](https://www.terraform.io/docs/cloud/api/changelog.html)

### CODE REVIEW

File: content/source/docs/cloud/api/providers.html.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+ `POST /organizations/:organization_name/registry-providers`
+
+Parameter            | Description
+---------------------|------------
+`:organization_name` | The name of the organization to create a provider in. The organization must already exist, and the token authenticating the API request must belong to the "owners" team or a member of the "owners" team.
+
+[permissions-citation]: #intentionally-unused---keep-for-maintainers
+
+#### Public providers
+When created, the public provider record will be available in the organization's registry provider list. You cannot create versions for public providers as they are maintained in the public registry.
```
Comment (laurapacilio) [inline content/source/docs/cloud/api/providers.html.md:175]:
```suggestion
When created, the public provider record will be available in the organization's registry provider list. You cannot create versions for public providers; you must use the versions available on the Terraform Registry.
```

File: content/source/docs/cloud/api/providers.html.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+* `DELETE /organizations/:organization_name/registry-providers/:registry_name/:namespace/:name`
+
+
+### Parameters
+
+Parameter            | Description
+---------------------|------------
+`:organization_name` | The name of the organization to delete a provider from. The organization must already exist, and the token authenticating the API request must belong to the "owners" team or a member of the "owners" team.
+`:namespace`         | The provider namespace that the deletion will affect.
+`:name`              | The provider name that the deletion will affect.
```
Comment (laurapacilio) [inline content/source/docs/cloud/api/providers.html.md:334]:
```suggestion
`:namespace`         | The namespace of the provider that will be deleted.
`:name`              | The name of the provider that will be deleted.
```

File: content/source/docs/cloud/api/providers.html.md [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+`:registry_name`     | `public`
+
+[permissions-citation]: #intentionally-unused---keep-for-maintainers
+
+
+Status  | Response                                             | Reason
+--------|------------------------------------------------------|-------
+[204][] | Nothing                                              | Success
+[403][] | [JSON API error object][]                            | Forbidden - public provider curation disabled
+[404][] | [JSON API error object][]                            | Provider not found or user not authorized
```
Comment (laurapacilio) [inline content/source/docs/cloud/api/providers.html.md:344]:
```suggestion
[404][] | [JSON API error object][]                            | Provider not found or user not authorized to perform action
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Hi! These are great already! I left a couple of suggestions. Let me know what you think - thank you!

**Review by laurapacilio — APPROVED** [review]
Approving so I don't block :-) We can press forward without my previous comments if necessary (though I do still have those recommendations!)

File: content/source/docs/cloud/api/providers.html.md [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+## Terraform Cloud Registry Implementation
+
+For publicly curated providers, the Terraform Cloud Registry acts as a proxy to the [Terraform Registry](https://registry.terraform.io) for the following:
+
+- List available versions for a specific provider
+- Get a specific provider
+- Get the latest version for a specific provider
+
+- The public registry discovery endpoints have the path prefix provided in the [discovery document](../../registry/api.html#service-discovery) which is currently `/api/registry/public/v1`.
+- [Authentication](./index.html#authentication) is handled the same as all other Terraform Cloud endpoints.
```
[the remark this answers is not in the document]
Comment (bethanyr) [reply content/source/docs/cloud/api/providers.html.md]: I went ahead and deleted for now, we don't publish the public registry provider endpoints.

File: content/source/docs/cloud/api/providers.html.md [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+
+## List Registry Providers for an Organization
+
+`GET /organizations/:organization_name/registry-providers`
+
+Parameter            | Description
+---------------------|------------
+`:organization_name` | The name of the organization to list available providers from.
+
+Lists the providers that are available to a given organization.
```
[the remark this answers is not in the document]
Comment (bethanyr) [reply content/source/docs/cloud/api/providers.html.md]: thanks - I have updated.

File: content/source/docs/cloud/api/providers.html.md [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+[200][] | [JSON API document][] (`type: "registry-providers"`) | The request was successful
+[404][] | [JSON API error object][]                          | Providers not found or user unauthorized to perform action
+
+### Query Parameters
+
+This endpoint supports pagination [with standard URL query parameters](./index.html#query-parameters); remember to percent-encode `[` as `%5B` and `]` as `%5D` if your tooling doesn't automatically encode URLs.
+
+Parameter                           | Description
+------------------------------------|---------------------
+`q`                                 | **Optional.** A search query string.  Providers are searchable by name, namespace fields.
```
[the remark this answers is not in the document]
Comment (bethanyr) [reply content/source/docs/cloud/api/providers.html.md]: thanks - I have updated.

**Pull request merged**